### PR TITLE
ref: Drop old/unused tag key logic

### PR DESCRIPTION
**Pull request by bretthoerner**
## Summary
- Adds `userCount` definition to `issue_properties_map` in the attributes reference
- Updates the issue search agent system prompt with user count in capabilities, examples, and basic query classification
- Adds `userCount` to test fixtures for query parser and tools tests

This teaches the AI search agent that `userCount:>N` is the correct syntax for filtering by number of affected users, preventing misinterpretation as `user.id` filters.

Companion sentry PR: https://github.com/getsentry/sentry/pull/114814

## Test plan
- [x] All 139 query parser tests pass
- [x] All 22 issues tools with parser tests pass

### CODE REVIEW

File: src/sentry/models/tagvalue.py [hunk 1]
```diff
@@ -49,35 +48,6 @@ def is_valid_value(cls, value):
...
-        if self.key == 'sentry:user':
-            return self.data.get('email') or self.value
-        elif self.key == 'sentry:function':
-            return '%s in %s' % (self.data['function'], self.data['filename'])
-        elif self.key == 'sentry:filename':
-            return self.data['filename']
-        elif self.key == 'sentry:release' and len(self.value) == 40:
-            return self.value[:12]
+        if self.key == 'sentry:release':
+            return Release.get_display_version(self.value)
```
Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: 🥇

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
Since you broke this, can we add a test to prove that it's not broken now and won't in the future? :)

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by bretthoerner** [comment]
Ya *I guess*. Test added.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**